### PR TITLE
add disjoint_set using list instead of node

### PR DESCRIPTION
**Pull request by JadeKim042386**
### Describe your change:

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/disjoint_set/disjoint_set_2.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+n = 9
```
Comment (rohan472000) [inline data_structures/disjoint_set/disjoint_set_2.py:1]: No docs is provided for this PR, you should look into other merged PRs for docs syntax.

File: data_structures/disjoint_set/disjoint_set_2.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+parent = list(range(n))
+
+rank = [0] * n
+
+
+def union_set(x: int, y: int) -> None:
```
Comment (rohan472000) [inline data_structures/disjoint_set/disjoint_set_2.py:8]: The `union_set` function has a logic error. It checks if x is equal to y before performing any operations, but it should be checking if `find_set(x)` is equal to `find_set(y)`. Replace `if x == y:` with `if find_set(x) == find_set(y):`.

File: data_structures/disjoint_set/disjoint_set_2.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    Union of two sets.
+    set with bigger rank should be parent, so that the
+    disjoint set tree will be more flat.
+    """
```
Comment (rohan472000) [inline data_structures/disjoint_set/disjoint_set_2.py:13]: Add doctests for every function.

File: data_structures/disjoint_set/disjoint_set_2.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            else:
+                assert find_set(node0) == find_set(node1)
+
+    # compressed path
+    print("parent:", parent)
+    print("rank:", rank)
+
+
+if __name__ == "__main__":
+    test_disjoint_set()
```
Comment (rohan472000) [inline data_structures/disjoint_set/disjoint_set_2.py:91]: add this in main - `doctest.testmod()` after adding doctests in every function

**Review by rohan472000 — CHANGES_REQUESTED** [review]

File: data_structures/disjoint_set/disjoint_set_2.py (hunk 4 above)
Comment (madhavgaba) [reply data_structures/disjoint_set/disjoint_set_2.py:91]: +1

Comment (888lilili) [reply data_structures/disjoint_set/disjoint_set_2.py:91]: 11111

**Review by tianyizheng02 — COMMENTED** [review]
I don't feel that this adds much value over the existing implementations. [data_structures/disjoint_set/alternate_disjoint_set.py](https://github.com/TheAlgorithms/Python/blob/9d86d4edaa754af06e0da9cac4a717f3765db7f4/data_structures/disjoint_set/alternate_disjoint_set.py) already implements disjoint set using lists, and this implementation specifically is a near-copy of [data_structures/disjoint_set/disjoint_set.py](https://github.com/TheAlgorithms/Python/blob/9d86d4edaa754af06e0da9cac4a717f3765db7f4/data_structures/disjoint_set/disjoint_set.py):
```py
# This PR
def union_set(x: int, y: int) -> None:
    """
    Union of two sets.
    set with bigger rank should be parent, so that the
    disjoint set tree will be more flat.
    """
    x, y = find_set(x), find_set(y)
    if x == y:
        return

    if rank[x] > rank[y]:
        parent[y] = x
    else:
        parent[x] = y
        if rank[x] == rank[y]:
            rank[x] += 1  # Also, I think this might be a typo in the PR
```
```py
# data_structures/disjoint_set/disjoint_set.py
def union_set(x: Node, y: Node) -> None:
    """
    Union of two sets.
    set with bigger rank should be parent, so that the
    disjoint set tree will be more flat.
    """
    x, y = find_set(x), find_set(y)
    if x == y:
        return

    elif x.rank > y.rank:
        y.parent = x
    else:
        x.parent = y
        if x.rank == y.rank:
            y.rank += 1
```
@cclauss Do you think this change from Node to list adds enough value to consider merging this PR?

**Comment by JadeKim042386** [comment]
@tianyizheng02 
As you said, it's not worth adding a new one when i look back at the comments after some time. If you think other reviewers are unnecessary, you may do close this pr.

**Pull request closed without merging**